### ISSUE TITLE
Knex does not escape single quote in comment function

### ISSUE THREAD

**Issue by emilwihlander** [knex/knex#4081]
# Environment

Knex version: 0.21.6
Database + version: MySQL 8.0.22
OS: Windows 10 (Docker)

# Bug
When using the comment function the single quote (') isn't escaped and therefore makes the query invalid.

When running the following command:
```
knex.schema.createTable('test', t => {
  t.text('column1').primary().comment("The table's first column")
  t.text('column2').comment("The table's second column")
})
  .then(r => console.log('success'))
  .catch(e => console.log(e))
```
with this knexfile:
```
const defaults = {
  client: 'mysql2',
  version: process.env.MYSQL_VERSION,
  connection: {
    host: process.env.MYSQL_HOST,
    user: 'root',
    password: process.env.MYSQL_ROOT_PASSWORD,
    database: process.env.MYSQL_DATABASE,
  },
  migrations: {
    tableName: 'knex_migrations'
  }
}
```
The following error is printed:
```
Error: 
create table `test` (
    `column1` text comment 'The table's first column',
    `column2` text comment 'The table's second column'
)
- You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 's first column', `column` text comment 'The table's second column')' at line 1
```

As seen, the single quote (') isn't escaped as \\' and therefore makes the query invalid. I haven't tried this with other databases but the insert function seems to escape the single quote when I tested.

Here's an example in runkit: https://runkit.com/emilwihlander/5f97014c4458ca001ad49a69

**Comment by kibertoad** [comment]
@wickedest This was broken in https://github.com/knex/knex/pull/4077. Could you please take a look?

**Comment by wickedest** [comment]
@kibertoad, nah I only changed the mssql dialect.  This is an existing issue in mysql.  The issue can be fixed [here](https://github.com/knex/knex/blob/master/lib/dialects/mysql/schema/columncompiler.js#L153) by escaping single quotes.

```diff
-    return comment && `comment '${comment}'`;
+    return comment && `comment '${comment.replace('\'', '\\\'')}'`;
```

**Comment by kibertoad** [comment]
@wickedest My apologies, only had a cursory glance at the issue. Thank you for pointing me in the right direction, even if the final fix ended up being slightly more elaborate to handle more of the edge cases.

**Comment by kibertoad** [comment]
Released in 0.21.8
